hpack/hpack.py

```diff
@@ -84,135 +85,25 @@ def _to_bytes(string):
-        (b'age', b''),
-        (b'allow', b''),
-        (b'authorization', b''),
-        (b'cache-control', b''),
-        (b'content-disposition', b''),
-        (b'content-encoding', b''),
-        (b'content-language', b''),
-        (b'content-length', b''),
-        (b'content-location', b''),
-        (b'content-range', b''),
-        (b'content-type', b''),
-        (b'cookie', b''),
-        (b'date', b''),
-        (b'etag', b''),
-        (b'expect', b''),
-        (b'expires', b''),
-        (b'from', b''),
-        (b'host', b''),
-        (b'if-match', b''),
-        (b'if-modified-since', b''),
-        (b'if-none-match', b''),
-        (b'if-range', b''),
-        (b'if-unmodified-since', b''),
-        (b'last-modified', b''),
-        (b'link', b''),
-        (b'location', b''),
-        (b'max-forwards', b''),
-        (b'proxy-authenticate', b''),
-        (b'proxy-authorization', b''),
-        (b'range', b''),
-        (b'referer', b''),
-        (b'refresh', b''),
-        (b'retry-after', b''),
-        (b'server', b''),
-        (b'set-cookie', b''),
-        (b'strict-transport-security', b''),
-        (b'transfer-encoding', b''),
-        (b'user-agent', b''),
-        (b'vary', b''),
-        (b'via', b''),
-        (b'www-authenticate', b''),
-    ]
 
     def __init__(self):
-        self.header_table = collections.deque()
-        self._header_table_size = 4096  # This value set by the standard.
+        self.header_table = HeaderTable()
         self.huffman_coder = HuffmanEncoder(
             REQUEST_CODES, REQUEST_CODES_LENGTH
         )
 
-        # We need to keep track of whether the header table size has been
-        # changed since we last encoded anything. If it has, we need to signal
-        # that change in the HPACK block.
-        self._table_size_changed = False
-
     @property
     def header_table_size(self):
-        return self._header_table_size
+        return self.header_table.maxsize
 
     @header_table_size.setter
     def header_table_size(self, value):
-        log.debug(
-            "Setting header table size to %d from %d",
-            value,
-            self._header_table_size
-        )
-
-        # If the new value is larger than the current one, no worries!
-        # Otherwise, we may need to shrink the header table.
-        if value < self._header_table_size:
-            current_size = header_table_size(self.header_table)
-
-            while value < current_size:
-                header = self.header_table.pop()
-                n, v = header
-                current_size -= (
-                    32 + len(n) + len(v)
-                )
-
-                log.debug(
-                    "Removed %s: %s from the encoder header table", n, v
-                )
-
-        if value != self._header_table_size:
-            self._table_size_changed = True
-
-        self._header_table_size = value
+        self.header_table.maxsize = value
```

Comment:
We'd like to keep hold of the two logs that used to be in this function as well.